diff --git a/node_modules/remark-copy-linked-files/index.js b/node_modules/remark-copy-linked-files/index.js
index 409eebb..f83e190 100644
--- a/node_modules/remark-copy-linked-files/index.js
+++ b/node_modules/remark-copy-linked-files/index.js
@@ -135,6 +135,31 @@ module.exports = (opts = {}) => {
       definition: (...args) => handlers.url(...args),
       image: (...args) => handlers.url(...args),
       jsx: (...args) => handlers.html(...args),
+      mdxBlockElement: async (node) => {
+        const selectors = {
+          'img': ['src'],
+          'source': ['src'],
+          'video': ['poster', 'src'],
+          'audio': ['src'],
+          'param': ['value'],
+          'a': ['href'],
+        };
+
+        if (Object.keys(selectors).includes(node.name) && node.attributes) {
+          for (let attr of node.attributes) {
+            if (selectors[node.name].includes(attr.name)) {
+              
+              const asset = await handleUrl(attr.value);
+
+              assets.push(asset);
+              attr.value = asset ? asset.url : attr.value;
+            }
+          }
+        }
+
+        return node;
+      },
+      mdxSpanElement: (...args) => handlers.mdxBlockElement(...args),
     };
 
     const newTree = await map(tree, async (node) => {
